File: examples/shapes/implementations/ruby/test/test_triangle.rb
# frozen_string_literal: true

require 'test_helper'

# triangle
class TestTriangle < Minitest::Test
  def setup
    @triangle = Triangle.new(3, 4, 5)
  end

  # Polytest Suite: triangle

  # Polytest Group: polygon

  def test_edge_count
    assert_equal 3, @triangle.edge_count
  end

  def test_vertex_count
    assert_equal 3, @triangle.vertex_count
  end

  # Polytest Group: shape

  def test_perimeter
    assert_equal @triangle.side_a + @triangle.side_b + @triangle.side_c, @triangle.perimeter
  end

  def test_area
    s = (@triangle.side_a + @triangle.side_b + @triangle.side_c) / 2.0
    expected_area = Math.sqrt(
      s * (s - @triangle.side_a) * (s - @triangle.side_b) * (s - @triangle.side_c)
    )
    assert_equal expected_area, @triangle.area
  end
end

File: examples/shapes/implementations/python/tests/test_triangle.py
import pytest
from .. import Triangle

# Polytest Suite: triangle

@pytest.fixture
def triangle():
    return Triangle(3, 4, 5)

# Polytest Group: polygon

@pytest.mark.group_polygon
def test_edge_count(triangle):
    """A polygon should accurately count the number of edges it contains"""
    assert triangle.edge_count() == 3

@pytest.mark.group_polygon
def test_vertex_count(triangle):
    """A polygon should accurately count the number of verticies it contains"""
    assert triangle.vertex_count() == 3

# Polytest Group: shape

@pytest.mark.group_shape
def test_perimeter(triangle):
    """A shape should be able to accurately calculate its perimeter (or circumference)"""
    assert triangle.perimeter() == triangle.side_a + triangle.side_b + triangle.side_c

@pytest.mark.group_shape
def test_area(triangle):
    """A shape should be able to accurately calculate its area"""
    s = (triangle.side_a + triangle.side_b + triangle.side_c) / 2
    expected_area = (s * (s - triangle.side_a) * 
                    (s - triangle.side_b) * 
                    (s - triangle.side_c)) ** 0.5
    assert triangle.area() == expected_area